### ISSUE TITLE
Dependency conflict for `r` that does not exist on macOS

### ISSUE THREAD

**Issue by izaakm** [Homebrew/linuxbrew-core#12119]
I already installed a version of `rename` with homebrew, but `util-linux` (a dependency of `r`) needs to overwrite the link. This already has an issue #1526 (#848), but it's closed and the conversation is limited. How should I proceed in this case?

**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, Homebrew/homebrew-core formula (not cask) on macOS? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a Linux problem please file this issue at https://github.com/Homebrew/linuxbrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue? \* see below

To help us debug your issue please explain:

- What you were trying to do (and why)

Install r `brew install r`

- What happened (include command output)

The dependency `util-linux` installs `rename` and then links `rename`. I already have `rename` installed, so there is an error. I unlinked `rename` to continue the install, but now there's the conflict of relinking `rename`:

```
$ brew link rename
Linking /home/jmill/.linuxbrew/Cellar/rename/1.600_1...
Error: Could not symlink bin/rename
Target /home/jmill/.linuxbrew/bin/rename
is a symlink belonging to util-linux. You can unlink it:
  brew unlink util-linux

To force the link and overwrite all conflicting files:
  brew link --overwrite rename

To list all files that would be deleted:
  brew link --overwrite --dry-run rename
```

I plan to overwrite the link from `util-linux`, but I'm afraid I may run into additional problems in the future. It seems like homebrew should be able to handle this dependency issue (?).

- What you expected to happen

Install r without dependency conflicts.

- Step-by-step reproduction instructions (by running `brew install` commands)

```
brew install rename
# Be sure that rename is linked
brew install r
```

\* 
```
brew config; brew doctor
HOMEBREW_VERSION: 2.0.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 417a16fbafa97a5740d355ed33bfbfd2823452e7
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/linuxbrew-core.git
Core tap HEAD: 693e6f41a6eaa2f5c26a709f21f134bb70fb7e1a
Core tap last commit: 9 hours ago
HOMEBREW_PREFIX: /home/jmill/.linuxbrew
HOMEBREW_REPOSITORY: /home/jmill/.linuxbrew/Homebrew
HOMEBREW_CELLAR: /home/jmill/.linuxbrew/Cellar
HOMEBREW_CACHE: /home/jmill/.cache/Homebrew
HOMEBREW_LOGS: /home/jmill/logs/linuxbrew
HOMEBREW_SYSTEM_CURL_TOO_OLD: 1
CPU: 64-core 64-bit unknown_0x15_0x2
Homebrew Ruby: 2.3.7 => /home/jmill/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/bin/ruby
Clang: N/A
Git: 2.21.0 => /home/jmill/.linuxbrew/bin/git
Curl: 7.64.0 => /home/jmill/.linuxbrew/opt/curl/bin/curl
Java: 1.7.0_181
Kernel: Linux 3.19.0-31-generic x86_64 GNU/Linux
OS: Ubuntu 14.04.5 LTS (trusty)
Host glibc: 2.19
/usr/bin/gcc: 4.8.4
glibc: 2.23
gcc: 5.5.0_4
xorg: 20170115_1

Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and what additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew provided
script of the same name. We found the following "config" scripts:
  /home/jmill/anaconda3/bin/pcre-config
  /home/jmill/anaconda3/bin/python3-config
  /home/jmill/anaconda3/bin/python3.6-config
  /home/jmill/anaconda3/bin/libpng-config
  /home/jmill/anaconda3/bin/icu-config
  /home/jmill/anaconda3/bin/krb5-config
  /home/jmill/anaconda3/bin/curl-config
  /home/jmill/anaconda3/bin/ncurses6-config
  /home/jmill/anaconda3/bin/freetype-config
  /home/jmill/anaconda3/bin/python3.6m-config
  /home/jmill/anaconda3/bin/libpng16-config
  /home/jmill/anaconda3/bin/xslt-config
  /home/jmill/anaconda3/bin/ncursesw6-config
  /home/jmill/anaconda3/bin/xml2-config
  /home/jmill/anaconda2/bin/libpng-config
  /home/jmill/anaconda2/bin/icu-config
  /home/jmill/anaconda2/bin/curl-config
  /home/jmill/anaconda2/bin/freetype-config
  /home/jmill/anaconda2/bin/python-config
  /home/jmill/anaconda2/bin/libpng16-config
  /home/jmill/anaconda2/bin/libdynd-config
  /home/jmill/anaconda2/bin/xslt-config
  /home/jmill/anaconda2/bin/xml2-config

Warning: You have unlinked kegs in your Cellar
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  openjdk
  rename

Warning: Your Homebrew's prefix is not /home/linuxbrew/.linuxbrew.
Some of Homebrew's bottles (binary packages) can only be used with the default
prefix (/home/linuxbrew/.linuxbrew).
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Discourse, Twitter or IRC. You are responsible for resolving any issues you
experience, as you are running this unsupported configuration.
```

**Comment by iMichka** [comment]
`brew uninstall rename` I guess. Both install a `rename` binary, you can't have both installed at the same time.

**Comment by iMichka** [comment]
Closing as I proposed a workaround. Let me know if this works for you.

**Comment by izaakm** [comment]
Thanks for the suggestion. I checked the `rename` utility installed by `util-linux` and it seems to be a distinct program from `rename` available directly from homebrew. 

I used the suggestion in the error message to overwrite the `rename` utility from `util-linux` with the previous `rename` program I installed:

```
To force the link and overwrite all conflicting files:
  brew link --overwrite rename
```

I don't use R on this machine, it's a dependency of some other software that I'm using. I'll just have to wait and see if something breaks to know if there's a conflict.
